src/modbus.c

```diff
@@ -1399,6 +1408,35 @@ int modbus_write_registers(modbus_t *ctx, int addr, int nb, const uint16_t *src)
     return rc;
 }
 
+int modbus_read_device_information_object(modbus_t *ctx, unsigned char object_id, char* dest)
+{
+    uint8_t req[_MIN_REQ_LENGTH];
+    uint8_t rsp[MAX_MESSAGE_LENGTH];
+    int rc;
+    int req_length;
+    int offset;
+    int i;
+    req_length = ctx->backend->build_request_basis(ctx, MODBUS_FC_READ_DEVICE_INFORMATION, 0x0E04, (object_id<<8), req) - 1;
+    rc = send_msg(ctx, req, req_length);
+    if (rc > 0) {
+        rc = _modbus_receive_msg(ctx, rsp, MSG_CONFIRMATION);
+        if (rc == -1)
+            return -1;
+
+        rc = check_confirmation(ctx, req, rsp, rc);
+        if (rc == -1)
+            return -1;
+
+        offset = ctx->backend->header_length + 9;
+
+        for (i = 0; i < rc; i++) {
+            dest[i] = rsp[offset + i];
+        }
+        dest[i] = '\0';
```

Comment:
this looks dubious.  While objects 1..6 are ascii per spec, the rest are all user defined.  Should this function not just return the length as is done in many other libmodbus functions?